### ISSUE TITLE
More realistic Cabin Ready system

### ISSUE THREAD

**Issue by Samueleonard** [flybywiresim/a32nx#6208]
### Aircraft Version

Experimental

### Description

Currently the only way to trigger a cabin-ready message is to press the call button, which instantly triggers the cabin-ready memo. This is obviously very unrealistic and so I propose this method instead.

**DEPARTURE**
. on departure/aircraft load?, a value is set for the number of seconds that it will take for the cabin to be ready (this number could suffice for departure and arrival or be 2 separate values?)
. once the first engine has begun to start (the point where the cabin would begin any safety demonstrations etc) the timer decreases
. once the timer hits 0, the cabin is ready and an ACP call is triggered or ecam memo is shown (described below)
OPTION - by pressing the call buttons, the pilot can reduce this length of time, thus speeding up the cabin, however, this can only be done once to prevent people spamming to instantly ready the cabin.

**ARRIVAL**
. once the seatbelt sign is turned back on, and the cabin would begin to prepare the cabin for landing, the same situation would apply as above (the counter begins to move down)
. for example short flights, if the seatbelt sign is never turned off, then the cabin would not need to be reprepared? maybe adjust depending on length of time that the belts were off for?

**EXTRA**
. with the merging of #6175 it should now be easier to prevent the time being changed to speed up the cabin timer.
. in terms of realistic values for the cabin to be ready, you could consult with pilots/cabin crew for durations at different passenger loads
. all of the above could be further tweaked to be dependant on your passenger loads. For example - heavier aircraft with more passengers would take longer to be ready.
. rather than instantly triggering the memo, the purser could ring the cockpit through the ACP cabin, triggering the ringing sound in the cockpit, at which point you would select the CAB channel, and the cabin would tell you that they are ready, as is done in real life.
. ALTERNATIVELY OR ADDITIONALLY, you could have an EFB option to have it triggered through the FAP, and appear as the ecam memo, as happens now, but with the additional logic as above.

### Discord Username (optional)

1samueleonard1

**Comment by tracernz** [comment]
There's a bit of work on this in https://github.com/flybywiresim/a32nx/pull/4997. If anyone wants to pick it up please do as I don't have time atm.

**Comment by beheh** [comment]
> by pressing the call buttons, the pilot can reduce this length of time, thus speeding up the cabin, however, this can only be done once to prevent people spamming to instantly ready the cabin.

> with the merging of #6175 it should now be easier to prevent the time being changed to speed up the cabin timer.

I don't think we should be concerned about this. If the user wants to simulate a time jump (e.g. simulating the aircraft standing there 5 minutes) why should we try so hard to prevent that? Similarly, while I can imagine the cabin speeding up when pinged, if the user presses it 5 times in a row they've pretty clearly indicated their intent to skip this.

**Comment by Samueleonard** [comment]
> > by pressing the call buttons, the pilot can reduce this length of time, thus speeding up the cabin, however, this can only be done once to prevent people spamming to instantly ready the cabin.
> 
> > with the merging of #6175 it should now be easier to prevent the time being changed to speed up the cabin timer.
> 
> I don't think we should be concerned about this. If the user wants to simulate a time jump (e.g. simulating the aircraft standing there 5 minutes) why should we try so hard to prevent that? Similarly, while I can realistically imagine the cabin speeding up only so much when pinged, if the user presses it 5 times in a row they've pretty clearly indicated their intent to skip this.

personally, i disagree.

this is being added in as an extra way to increase realism, so it should be kept as realistic as possible imo.
if someone doesnt want to use the feature (spamming the call button to skip) then they could just depart without the cabin being ready

**Comment by sidnov** [comment]
Partially covered in #6732
